lib/helpers/utils.js

```diff
@@ -10869,7 +10869,25 @@ export function parseCsProjData(csProjData, projFile, pkgNameVersions = {}) {
           continue;
         }
         pkg.name = pref.Include;
-        pkg.version = pref.Version || pkgNameVersions[pkg.name];
+        pkg.version =
+          pref.Version ||
+          item.PackageReference[j].Version?.[0]._?.[0] ||
+          pkgNameVersions[pkg.name];
+        if (pkg.version) {
+          const version_label_match = pkg.version.match(/^\$\((.*)\)$/);
+          if (version_label_match) {
+            const result = safeSpawnSync(
+              DOTNET_CMD,
+              [
+                `msbuild ${projFile} -nologo -getProperty:${version_label_match[1]}`,
```

Comment:
Would this invoke msbuild for all projects including non-windows builds? Can we instead parse the props files and try and substitute values first?